v19.2/api-stability.md

```diff
@@ -0,0 +1,586 @@
+  to the stability guarantees**](#exceptions-to-stability-guarantees)
+  below.
+
+## Definitions
+
+### Major version and patch revision
+
+Major versions are identified by the first two numbers in the full
+version string, and patch revisions are identified by the last number
+in the string.
+
+For example:
+
+| Version string | Major version | Patch revision |
+|----------------|---------------|----------------|
+| 2.1            | 2.1           | 0              |
+| 2.1.4          | 2.1           | 4              |
+| 19.1           | 19.1          | 0              |
+| 19.1.2         | 19.1          | 2              |
+
+### Public and programmable interfaces
+
+These interfaces are meant for interfacing with third party automated
+tools.
+
+For example, the output of a `SELECT` query is public and
+programmable.
+
+For these interfaces, an app/client that works with major version N is expected
+to work with all patch revisions. Compatibility with version N+1 and
+later depends on the stability phase, as detailed below in section
+[Stability phases for public and programmable
+interfaces](#stability-phases-for-public-and-programmable-interfaces).
+
+Additionally, the CockroachDB team commits to documenting these
+interfaces over time, and/or upon request.
+
+### Public and non-programmable interfaces
+
+These interfaces are meant to be consumed by humans and
+are not suitable for automation.
+
+For example, the output of SQL `SHOW` statements
+is public but non-programmable.
+
+For these interfaces, the stability guarantees are as follows:
+
+- the particular data format of the inputs and outputs may change across patch
+  revisions, although the CockroachDB team will make some effort to avoid this;
+- the particular data format of the inputs and outputs are likely to change
+  across major versions.
+
+These interfaces may not be documented, but can be incidentally
+explained on-demand by the CockroachDB team during discussions on
+public forums or during troubleshooting sessions. Users are invited
+to reuse this knowledge.
+
+### Reserved interfaces
+
+These interfaces are meant for use by CockroachDB developers and are not
+suitable for use, neither in automation by 3rd parties, nor by external documentation
+that aims to remain relevant across CockroachDB revisions.
+
+For example, the contents of certain tables in the `crdb_internal` SQL
+virtual schema is a reserved interface.
+
+For these, the stability guarantees are as follows:
+
+- the particular data format of inputs or outputs may change between
+  patch revisions.
+- there is no expectation of accurate or up-to-date documentation.
+
+## Stability phases for public and programmable interfaces
+
+New features or behavior changes in public/programmable interfaces are
+typically released first in status “experimental” then goes to “beta”
+then evolves to “stable”.
+
+Unless specified otherwise, a new feature that appears in version N in
```

Comment:
I am reading this sentence as "everything in a new release is beta unless we explicitly document that it is stable".

However, our docs do not appear to currently follow this rule.  AFAICT, we do not explicitly document anywhere that a feature is stable.  We only explicitly note when a feature is unstable (beta/experimental), and remove the note when the feature graduates to stable.

Am I totally misreading this sentence? Please enlighten me if so.

Reply:
I think the intended meaning is "if the docs are silent, assume that the first release of a new feature is beta and that is automatically gets stabilized in the next release". Of course if the docs are explicit, that supersedes.

Reply:
Got it - thanks for explaining.